src/ui/src/renderer_widget.cpp

```diff
@@ -82,3 +94,52 @@ QSurfaceFormat surface_format(const QSurfaceFormat::FormatOptions options) {
 	surface_format.setStencilBufferSize(8);
 	return surface_format;
 }
+
+void RendererWidget::mousePressEvent(QMouseEvent* event) { 
+	previous_mouse_pos_ = event->pos();
+	event->accept();
+}
+
+
+void RendererWidget::mouseReleaseEvent(QMouseEvent* event) { 
+	event->accept(); 
+}
+
+void RendererWidget::mouseMoveEvent(QMouseEvent* event) {
+	auto clicked_buttons = event->buttons();
+	// if button still pressed we use it and don't mention other clicked buttons
+	if (current_pressed_button_ & clicked_buttons)
+		clicked_buttons = current_pressed_button_;
+	bool action_requiered = true;
+	const auto q_curr_pos = event->pos();
+	const auto q_delta = q_curr_pos - previous_mouse_pos_;
+	const Vec2D curr_pos(q_curr_pos.x(), q_curr_pos.y());
```

Comment:
Vec2D is simple struct with no explicitly defined ctors. this code is compiling on gcc and msvc, but will not on clang. Also it's just misinformation about what it is. Use curly braces initialization (with or without ` = `)